wcs-importer-exporter.php

```diff
@@ -61,10 +64,31 @@ public static function setup_importer() {
 	}
 
 	/**
+	 * This is to calculate last_payment when the last payment not set by import
+	 * this is done by taking next interval and removing twice.
+	 * Only change from 0 if it was an imported subscription
+	 *
+	 * @param $date
+	 * @param $subscription @class WC_Subscription
+	 * @param $timezone
+	 * @return $date or date of last payment calulation
+	 */
+	public static function last_payment_calculation($date, $subscription, $timezone ) {
+		if ($date == 0 && "importer" === $subscription->__get("created_via")) {
+			$next_payment = strtotime($subscription->get_date('next_payment'));
+			$next_interval = wcs_add_time( $subscription->billing_interval, $subscription->billing_period, $next_payment );
+			$last_payment_cal = $next_payment - ($next_interval - $next_payment);
+			return  gmdate( 'Y-m-d H:i:s',$last_payment_cal);
+		}
+		return $date;
```

Comment:
Because the `$timezone` is being ignored here, if someone is requesting `$subscription->get_date( 'last_payment', 'site' )`, this will return an incorrect date. It needs to also check `'gmt' != strtolower( $timezone )` and when that's the case, it should change the `$last_payment_cal` value to be in the site's timezone using something like:

```
$last_payment_cal = get_date_from_gmt( $last_payment_cal )
```